### PR TITLE
detect/ipopts: IP option tests

### PR DESCRIPTION
**Pull request by jlucovsky**
## Ticket

Redmine ticket: [6864](https://redmine.openinfosecfoundation.org/issues/6864)

Add tests for IP options lacking coverage -- "extended security" is tested by `ipopts-esec`

From the README:
Test the IP options. There's already a test for the extended security option; the following IP options are
tested:
- Record Route "rr"
- Loose source route "lsrr"
- EOL "eol"
- NOP "nop"
- Timestamp "ts"
- Security "sec"
- Strict source route "ssrr"
- Stream id "satid"

The pcap was generated using ipopt.py

### CODE REVIEW

**Comment by catenacyber** [comment]
This requires https://github.com/OISF/suricata/pull/10688 right ?

Should the first SV PR only fail master ?

File: tests/detect-ipopts/test.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+args:
+  - --set stream.midstream=true -k none
+
+checks:
+  - filter:
+      count: 1
+      match:
+        event_type: alert
+        alert.signature_id: 1
```
Comment (catenacyber) [inline tests/detect-ipopts/test.yaml:9]: Should we test pcap_cnt to see that each alert matches act the right packet ?

**Comment by victorjulien** [comment]
This PR failing on all branched currently blocks merge of https://github.com/OISF/suricata/pull/10859

**Comment by jlucovsky** [comment]
> This PR failing on all branched currently blocks merge of [OISF/suricata#10859](https://github.com/OISF/suricata/pull/10859)

I've added a version qualifier for the tests. It'll fail on unmodified master and will be skipped on main-7.0.x and master-6.0.x

File: tests/detect-ipopts/test.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        pcap_cnt: 1
+        alert.signature_id: 1
+  - filter:
+      count: 1
+      match:
+        event_type: alert
+        pcap_cnt: 2
+        alert.signature_id: 2
+  - filter:
+      count: 6
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply tests/detect-ipopts/test.yaml:21]: Oh, end of list (of options)

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #1780, thanks!

**Pull request closed without merging**